test/CMakeLists.txt

```diff
@@ -46,9 +50,9 @@ set_target_properties(GreeterTests PROPERTIES CXX_STANDARD 17)
 # enable compiler warnings
 if(NOT TEST_INSTALLED_VERSION)
   if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID MATCHES "GNU")
-    target_compile_options(Greeter PUBLIC -Wall -pedantic -Wextra -Werror)
+    target_compile_options(Greeter PRIVATE -Wall -pedantic -Wextra -Werror)
   elseif(MSVC)
-    target_compile_options(Greeter PUBLIC /W4 /WX)
+    target_compile_options(Greeter PRIVATE /W4 /WX)
```

Comment:
`PUBLIC` is better imo, as the warnings will be applied to the test target as well. This is especially useful if testing code defined in headers.

Reply:
I do not disable this, but it should not exported!
it would break other project which are not free of warnings ...

Reply:
Exactly, that is why we only add the flags when compiling the tests and not when using it as a library.

Reply:
OK